File: zoj_acm/24/2420.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

DayOfWeek = {1: 'Sunday',
             2: 'Monday',
             3: 'Tuesday',
             4: "Wednesday",
             5: 'Thursday',
             6: 'Friday',
             0: 'Saturday'
             }


def proc(n):
    years = 2000
    days = 366
    month = 1
    days_month = 31
    while n > days:
        years += 1
        n = n - days
        if (years % 4 == 0 and years % 100 != 0) or (years % 400 == 0):
            days = 366
        else:
            days = 365
    if days == 366:
        flag = 1
    else:
        flag = 0
    while n > days_month:
        month += 1
        n = n - days_month
        if month in [1, 3, 5, 7, 8, 10, 12]:
            days_month = 31
        elif month in [4, 6, 9, 11]:
            days_month = 30
        elif flag == 1:
            days_month = 29
        else:
            days_month = 28

    return years, month, n


def zoj_2420():
    while True:
        line = int(raw_input())
        if line == -1:
            break
        else:
            years, months, days = proc(line + 1)
            week = DayOfWeek.get(line % 7)
            print "%d-%02d-%02d %s" % (years, months, days, week)

if __name__ == '__main__':
    zoj_2420()

File: zoj_acm/24/2420.cpp
#include<stdio.h>  
int main(){  
    char w[7][10]={"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday" ,"Saturday"};  
    int m[2][13]={0,31,28,31,30,31,30,31,31,30,31,30,31,  
                  0,31,29,31,30,31,30,31,31,30,31,30,31};  
    int yd[2]={365,366};  
    long day;  
    int year,month,week;  
    int i,j,flag;  
    while(scanf("%ld",&day)&&-1!=day){  
        week=(day+6)%7;//得到星期几   
        year=2000;  
        flag=(0==year%4&&year%100!=0)||0==year%400;//flag=1为闰年   
        ++day;//题目说经过多少天，所以在这里先加1   
        for(;day>yd[flag];){//得到年份、剩余天数   
            day-=yd[flag];   
            year++;  
            flag=(0==year%4&&year%100!=0)||0==year%400;  
        }  
        for(month=1;day>m[flag][month];++month){//得到月份和对应天数   
            day-=m[flag][month];  
        }  
        printf("%d-%02d-%02d %s\n",year,month,day,w[week]);//%02d很方便:如果整数不够2列就补上0          
    }  
}